app/controllers/ui_controller.rb

```diff
@@ -3,6 +3,8 @@ class UiController < ApplicationController
     redirect_to :root if Rails.env.production?
   end
 
+  before_action :require_user
```

Comment:
I'm not sure you need to lock this controller down, since it isn't available in production anyway.